### ISSUE TITLE
Is there a way to externally denoise the MLV before loading it into MLVapp? 

### ISSUE THREAD

**Issue by ciberetica** [ilia3101/MLV-App#239]
I'm trying to de-noise MLVs with ffpmeg (nlmeans works [really really well](https://dirk-farin.net/projects/nlmeans/index.html)) before editing with MLVapp. I don't use DaVinci or any other color grading editor, only MLVapp. I would like to be able to remove noise with nlmeans before adding sharpness and other adjustments with MLVapp, but I can't find a way... Is this possible? Thanks a lot.

An image to illustrate how well it works! 

![](https://forum.handbrake.fr/download/file.php?id=142&sid=0e847ad83d0a85bf999703643d686c6e)

**Comment by masc4ii** [comment]
Hi,
as long ffmpeg can't really handle ML RAW files, and I also don't know if it is possible to apply a RGB denoiser to RAW data at all, I think it will not be possible.
What you could do instead: apply sharpening after denoising also in ffmpeg! E.g. Handbrake does this, too, also with GUI.

**Comment by ciberetica** [comment]
> Hi, as long ffmpeg can't really handle ML RAW files, and I also don't know if it is possible to apply a RGB denoiser to RAW data at all, I think it will not be possible. What you could do instead: apply sharpening after denoising also in ffmpeg! E.g. Handbrake does this, too, also with GUI.

Good idea! I'll do that, edit as usual in MLVapp and then add the sharpening and denoise afterwards! I know that MLVapp is not a color editing program as such, but it works so well that I like it more than any other option.

Thank you very much for the advice! As long as there is nothing better, this can work for me.

**Comment by masc4ii** [comment]
MLVApp was intended to be a MLV color correction/grading processing tool. It is the main task of this application.

**Comment by ciberetica** [comment]
> 

I am sorry to have expressed myself badly, but I speak through a translator. I meant to be a grading processing tool "as such" (I don't know if this expression is correct) in the sense that you don't work with intermediate codecs and therefore the referenced denoise cannot be applied, for example.

Of course it is, and for me without a doubt the best. Not only because of the speed, simplicity or power, especially because it is free software and this is something important.
